From 3d6270de602bd2d12fc23ab537189de8a23362ab Mon Sep 17 00:00:00 2001
From: Victor Hsieh <victorhsieh@google.com>
Date: Fri, 4 Dec 2020 12:49:14 -0800
Subject: [PATCH] UPSTREAM: Android: correct statfs64.f_fsid type & visibility

The visibility has been not public since the initial commit[1]. But there
is no reason to hide this struct member specifically.

The type has also been "__fsid_t" in Android's bionic[2], so fix that as
well. This makes it consistent with statfs.f_fsid nicely.

[1] a36da11fb9cfcafcee6cb263bf1dc2c84371730a
[2] https://cs.android.com/search?q=file:bionic%2Flibc%20f_fsid

Bug: 174797066
Test: m
Change-Id: Ib4d88965d36fb5537b7df241e4a847e33efbd27a
---
 src/unix/linux_like/android/b32/mod.rs | 2 +-
 src/unix/linux_like/android/b64/mod.rs | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/unix/linux_like/android/b32/mod.rs b/src/unix/linux_like/android/b32/mod.rs
index 5c4f0323..e686da60 100644
--- a/src/unix/linux_like/android/b32/mod.rs
+++ b/src/unix/linux_like/android/b32/mod.rs
@@ -74,7 +74,7 @@ s! {
         pub f_bavail: u64,
         pub f_files: u64,
         pub f_ffree: u64,
-        f_fsid: [u32; 2],
+        pub f_fsid: ::__fsid_t,
         pub f_namelen: u32,
         pub f_frsize: u32,
         pub f_flags: u32,
diff --git a/src/unix/linux_like/android/b64/mod.rs b/src/unix/linux_like/android/b64/mod.rs
index 0f9443f1..419c2502 100644
--- a/src/unix/linux_like/android/b64/mod.rs
+++ b/src/unix/linux_like/android/b64/mod.rs
@@ -84,7 +84,7 @@ s! {
         pub f_bavail: u64,
         pub f_files: u64,
         pub f_ffree: u64,
-        f_fsid: [u32; 2],
+        pub f_fsid: ::__fsid_t,
         pub f_namelen: u64,
         pub f_frsize: u64,
         pub f_flags: u64,
-- 
2.29.2.684.gfbc64c5ab5-goog

